### DISCUSSION TITLE
How to rename the "app" directory and "router.options.ts" ?

### DISCUSSION THREAD

**Discussion by TokugawaTakeshi** [nuxt/nuxt#18954] [Questions]
For the post day, the "app" directory does not present in the table of contents of the documentation:

![image](https://user-images.githubusercontent.com/41653501/218240536-2df99902-d230-43b0-abbb-3c89b169bcb4.png)

I have knew about it only from [Custom Routes](https://nuxt.com/docs/guide/directory-structure/pages#custom-routes) section.

I want to move the `"router.options.ts"` to other directory and if it is possible - rename this file.
Can I do this?

**Comment by TheAlexLichter** [comment]
1. Feel free to create an issue pointing out that the folder is missing and also feel free to send a PR to add the `app` folder there 🙏🏻
2. No, neither renaming nor moving is possible right now. [It is hardcoded](https://github.com/nuxt/nuxt/blob/dda0cebc761c00042ad82bb4c2a47b679581755a/packages/nuxt3/src/pages/module.ts#L87).

**Reply by AloisSeckar** [reply]
I wouldn't mind preparing a draft for a new doc page about `/app` directory, but I am not sure about all the purposes it has.

I see it is also mentioned when [talking about layers](https://nuxt.com/docs/guide/going-further/layers#basic-example). Also an example project is [linked](https://nuxt.com/docs/getting-started/layers#examples).

Do you think simple mentioning those use cases with link to the respective places will be sufficient for the new page?

**Reply by TheAlexLichter** [reply]
> I wouldn't mind preparing a draft for a new doc page about /app directory, but I am not sure about all the purposes it has.

Let's discuss that in an issue 👍🏻
Maybe we don't need explicit docs as of right now.
The `app` folder contains several files which should override files provided by the default nuxt instance (in Nuxt2 e.g. the ssr error page, the base HTML file etc. etc.)

> I see it is also mentioned when [talking about layers](https://nuxt.com/docs/guide/going-further/layers#basic-example). Also an example project is [linked](https://nuxt.com/docs/getting-started/layers#examples).

That's a bit ambiguous and means a "normal" project folder. PR welcome for e.g. `my-app` there!

**Reply by AloisSeckar** [reply]
> That's a bit ambiguous and means a "normal" project folder. PR welcome for e.g. `my-app` there!

So, if I got it right from the [demo](https://github.com/nuxt/nuxt/tree/main/examples/advanced/config-extends):

(current) `app` is meant to be Nuxt project's root and `base` is some sub-folder with additional content? 

If yes, then I would also extend the other two file paths to `my-app/base/nuxt.config.ts` and `my-app/base/components/BaseComponent.vue` to make it clear for dummies like me. 

Or maybe, for the sakes of simplicity, just remove `app/` from the first two...

**Reply by AloisSeckar** [reply]
I allowed myself to purpose the PR for this minor change #18967 and also opened the issue for `app` page itself #18966

**Reply by TokugawaTakeshi** [reply]
@manniL 

Thank you for the answer.
With your permission, I [have opened the issue](https://github.com/nuxt/nuxt/issues/19081).
